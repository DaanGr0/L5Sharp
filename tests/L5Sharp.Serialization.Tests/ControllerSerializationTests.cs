using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using L5Sharp.Primitives;
using NUnit.Framework;

namespace L5Sharp.Primitives.Tests
{
    [TestFixture]
    public class ControllerSerializationTests
    {
        [Test]
        public void Serialize_WhenCalled_ShouldNotBeNull()
        {
            var controller = new Controller("ControllerName");
            
            var element = controller.Serialize();

            element.Should().NotBeNull();
        }

        
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Serialize_WhenCalled_ShouldHaveApprovedOutput()
        {
            var controller = new Controller("ControllerName");
            
            var element = controller.Serialize();

            Approvals.VerifyXml(element.ToString());
        }
        
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Serialize_WithDataTypes_ShouldHaveApprovedOutput()
        {
            var controller = new Controller("ControllerName");
            
            controller.Create().DataType("DT1", b =>
                b.HasDescription("Test Type 1")
                    .WithMember("TestMember", DataType.Bool));
            
            controller.Create().DataType("DT2", b =>
                b.HasDescription("Test Type 2")
                    .WithMember("TestMember", DataType.Dint));
            
            var element = controller.Serialize();

            Approvals.VerifyXml(element.ToString());
        }
    }
}