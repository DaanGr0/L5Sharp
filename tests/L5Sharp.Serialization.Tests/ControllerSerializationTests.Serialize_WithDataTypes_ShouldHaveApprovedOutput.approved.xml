<Controller Name="ControllerName" ProcessorType="" MajorRev="0" MinorRev="0">
  <DataTypes />
  <DataType Name="DT1" Family="NoFamily" Class="User">
    <Description><![CDATA[Test Type 1]]></Description>
    <Members>
      <Member Name="ZZZZZZZZZZDT10" DataType="SINT" Dimension="0" Radix="Decimal" Hidden="true" ExternalAccess="Read/Write" />
      <Member Name="TestMember" DataType="BOOL" Dimension="0" Radix="Decimal" Hidden="false" Target="ZZZZZZZZZZDT10" BitNumber="0" ExternalAccess="Read/Write" />
    </Members>
  </DataType>
  <DataType Name="DT2" Family="NoFamily" Class="User">
    <Description><![CDATA[Test Type 2]]></Description>
    <Members>
      <Member Name="TestMember" DataType="DINT" Dimension="0" Radix="Decimal" Hidden="false" ExternalAccess="Read/Write" />
    </Members>
  </DataType>
  <Programs />
  <Tasks />
</Controller>